frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx

```diff
@@ -5,95 +5,133 @@ import { ImageStreamImportsModel } from '@console/internal/models';
+    k8sCreate(ImageStreamImportsModel, importImage)
+      .then((imageStreamImport) => {
+        const status = _.get(imageStreamImport, 'status.images[0].status');
+        if (status.status === 'Success') {
+          const name = _.get(imageStreamImport, 'spec.images[0].from.name');
+          const image = _.get(imageStreamImport, 'status.images[0].image');
+          const tag = _.get(imageStreamImport, 'status.images[0].tag');
+          const isi = { name, image, tag, status };
+          const ports = getPorts(isi);
           setFieldValue('isSearchingForImage', false);
-        });
-    },
-    [setFieldValue, values.application.name, values.name, values.project.name],
-  );
+          setFieldValue('isi.name', name);
+          setFieldValue('isi.image', image);
+          setFieldValue('isi.tag', tag);
+          setFieldValue('isi.status', status);
+          setFieldValue('isi.ports', ports);
+          setFieldValue('image.ports', ports);
+          setFieldValue('image.tag', tag);
+          !values.name && setFieldValue('name', getSuggestedName(name));
+          !values.application.name &&
+            setFieldValue('application.name', `${getSuggestedName(name)}-app`);
+          // set default port value
+          const targetPort = _.head(ports);
+          targetPort && setFieldValue('route.targetPort', makePortName(targetPort));
+          setValidated(VALIDATED.success);
+        } else {
+          setFieldValue('isSearchingForImage', false);
+          setFieldValue('isi', {});
+          setFieldValue('isi.status', status.message);
+          setFieldValue('route.targetPort', null);
+          setValidated(VALIDATED.error);
+        }
+      })
+      .catch((error) => {
+        setFieldValue('isi', {});
+        setFieldValue('isi.status', error.message);
+        setFieldValue('isSearchingForImage', false);
+        setValidated(VALIDATED.error);
+      });
+  }, [setFieldValue, values.application.name, values.name, values.project.name, values.searchTerm]);
 
   const handleSave = (name: string) => {
     setNewImageSecret(name);
-    values.searchTerm && handleSearch(values.searchTerm);
+    values.searchTerm && handleSearch();
   };
 
+  const getHelpText = () => {
+    if (values.isSearchingForImage) {
+      return (
+        <span style={{ fontWeight: 'bold' }}>
+          <LoadingInline />
+        </span>
+      );
+    }
+    if (!values.isSearchingForImage && validated === VALIDATED.success) {
+      return (
+        <span style={{ fontWeight: 'bold', color: 'var(--pf-global--success-color--200)' }}>
+          <CheckCircleIcon /> Validated
+        </span>
+      );
+    }
+
+    return '';
+  };
+
+  const getHelpTextInvalid =
+    validated === VALIDATED.error ? (
+      <span style={{ color: 'var(--pf-global--danger-color--200)' }}>
+        <ExclamationCircleIcon />
+        &nbsp;
+        {values.searchTerm === '' ? 'Required' : values.isi.status}
+      </span>
+    ) : (
+      ''
```

Comment:
You should almost never have to equate something down to an empty string. Controlled input values is the only thing I can think of... but you're creating a DOM element, which a falsy value should do fine.

Ie:
```
  const invalidHelpText = validated === VALIDATED.error && (
    <span className="odc-...">
      <ExclamationCircleIcon />
      {' '}
      {values.searchTerm === '' ? 'Required' : values.isi.status}
    </span>
  );
```